### PR TITLE
finished task 3

### PR DESCRIPTION
**Pull request by licuevas98**
task#3: maked the browser to get current weather conditions for the location of the user

### CODE REVIEW

**Comment by agfonte** [comment]
.idea folder should not be included in commit, please remove it

File: src/components/NavPage.js [hunk 1]
```diff
@@ -1,12 +1,102 @@
...
+import React, {useEffect, useState} from "react";
 import NavBar from "./NavBar";
+import WeatherInfo from "./WeatherInfo";
+import Box from "@mui/material/Box";
+import Typography from "@mui/material/Typography";
+import Grid from "@mui/material/Grid";
+import {Image} from "@mui/icons-material";
 
 function NavPage() {
+    const key = "NQ6gDQ8Tb5v7rkrFyUH7yUunW5R2VRWh";
```
Comment (agfonte) [inline src/components/NavPage.js]: Never put sensitive data into a commit, please move this to an env variable

File: src/components/NavPage.js [hunk 2]
```diff
@@ -1,12 +1,102 @@
...
+        fetch(baseUrl + query).then((response => {
+            const resp = response.clone();
+
+            resp.json().then((data) => {
+                setWeather(data[0].WeatherText)
+                setTemperature(data[0].Temperature.Metric.Value)
+            })
+        }))
+
+        //res.push(data[0].WeatherText);
```
Comment (agfonte) [inline src/components/NavPage.js]: remove commented code please

File: src/components/NavPage.js [hunk 3]
```diff
@@ -1,12 +1,102 @@
...
+    const key = "NQ6gDQ8Tb5v7rkrFyUH7yUunW5R2VRWh";
+    const [cityKey, setCityKey] = React.useState("");
+    const [weather, setWeather] = React.useState("");
+    const [temperature, setTemperature] = React.useState("");
+    const [cityText, setCityText] = useState("");
+    const [cityChange, setCityChange] = useState("");
+    //const [flag, setFlag] = useState(false);
+
+    //getting weather details
+    const gettingWeather = () => {
```
Comment (agfonte) [inline src/components/NavPage.js]: This methods should not be on the NavPage, please move it to a different file that will be in charge of doing all the logic of getting data from the API and then use those methods please

**Review by agfonte — CHANGES_REQUESTED** [review]
General guidelines:
- Never attack sensitive data into the code
- Try following design patterns like MVC, please separate the logic from the view from the model
- Do not add into the codebase unnecessary folders/files like .idea or others

**Review by licuevas98 — COMMENTED** [review]
task#3: implemented feedback in the project

File: src/App.js [hunk 4]
```diff
@@ -1,11 +1,5 @@
...
-import {BrowserRouter, Route} from 'react-router-dom';
-import routes from "./routes";
 import MainPage from "./components/MainPage";
-import SignIn from "./components/SignIn";
-import {Routes, useRoutes} from "react-router-dom";
-import Link from "@mui/material/Link";
-import NavPage from "./components/NavPage";
 
 function App() {
     // const routing = useRoutes(routes);
```
Comment (agfonte) [inline src/App.js]: Commented code, please remove it

**Review by agfonte — CHANGES_REQUESTED** [review]
The apikey should not be an input of the user, should be read from the environment (as an environment variable) or any other secure mechanism for sensitive data.

Please remove the input for the apikey for the user and use this type of mechanism

**Review by agfonte — APPROVED** [review]
Great job!

**Pull request merged**